ChatCompletionCustomSkill/function_app.py

```diff
@@ -203,44 +207,47 @@ async def custom_skill(req: func.HttpRequest) -> func.HttpResponse:
-                try:
-                    # Prepare messages
-                    messages = prepare_messages(request_body, scenario, custom_prompts)
-                    
-                    # Prepare request payload
-                    request_payload = {
-                        "messages": messages,
-                        "temperature": config.temperature,
-                        "top_p": config.top_p,
-                        "max_tokens": config.max_tokens
-                    }
-                    
-                    # Call model with timeout
-                    async with asyncio.timeout(config.timeout):
-                        response = await client.complete(request_payload)
-                        response_text = response.choices[0].message.content
-                        
-                    # Format response
-                    response_values.append(format_response(request_body, response_text, scenario))
-                    
-                except asyncio.TimeoutError:
-                    logger.error(f"Timeout processing record {request_body.get('recordId')}")
-                    response_values.append({
-                        "recordId": request_body.get("recordId"),
-                        "errors": ["Request timeout"],
-                        "warnings": None,
-                        "data": None
-                    })
-                except Exception as e:
-                    logger.error(f"Error processing record {request_body.get('recordId')}: {e}")
-                    response_values.append({
-                        "recordId": request_body.get("recordId"),
-                        "errors": [str(e)],
-                        "warnings": None,
-                        "data": None
-                    })
+        api_key = os.getenv("AZURE_INFERENCE_CREDENTIAL")
+        endpoint = os.getenv("AZURE_CHAT_COMPLETION_ENDPOINT")
+        headers = {
+        "Content-Type": "application/json",
+        "api-key": api_key,
+        "Authorization": f"Bearer {api_key}"
+        }
+        for request_body in input_values:
+            try:
+                messages = prepare_messages(request_body, scenario, custom_prompts)
+                request_payload = {
+                    "messages": messages,
+                    "temperature": config.temperature,
+                    "top_p": config.top_p,
+                    "max_tokens": config.max_tokens
+                }
+                # Call model with timeout
+                async with asyncio.timeout(config.timeout):
+                    vanilla_response = requests.post(endpoint, headers=headers, json=request_payload)
+                    vanilla_response.raise_for_status()  # Will raise an HTTPError if the HTTP request returned an unsuccessful status code
+                    vanilla_response_json = vanilla_response.json()
+                    response_text = vanilla_response_json['choices'][0]['message']['content']

```

Comment:
Do we need some checks in there if the structure is not what's expected or are we fine with the generic except below?

Reply:
Good question... we don't really know the level of exception handling, validation, etc. that we'll need until we get much further along in the Native Skill arc. If there any gaps, I will address them!